src/routes/club/Proposal.js

```diff
@@ -28,15 +28,49 @@ router.get("/get/:id", verifyClub, async (req, res) => {
 
 
 router.post("/create", verifyClub, async (req, res) => {
-
+	new Proposal(req.body)
+		.then( async (newProposal) => {
+			if(!newProposal) {
+				return res.status(500).send({ success: false, message: "Something went wrong" });
+			}
+			if(newProposal.clubId.equals(req.session.userId)) {
+				await newProposal.save();
+				return res.json({ success: true, newProposal });
+			}
+			else {
+				return res.status(403).send({ success: false, message: "Forbidden" });
+			}
+		})
 });
 
 router.put("/edit/:id", verifyClub, async (req, res) => {
-
+	Proposal.findByIdAndUpdate(req.params.id, req.body)
+		.then((proposal) => {
+				if(!proposal) {
+					return res.status(500).send({ success: false, message: "Proposal not found" });
+				}
+				if(proposal.clubId.equals(req.session.userId)) {
+					return res.json({ success: true, message: "Proposal Updated" });
+				}
+				else {
+					return res.status(403).send({ success: false, message: "Forbidden" });
+				}
+		});
 });
 
 router.delete("/delete/:id", verifyClub, async (req, res) => {
-
+	Proposal.findByIdAndDelete(req.params.id)
+		.then((proposal) => {
+			if(!proposal) {
+				return res.status(500).send({ success: false, message: "Proposal not found" });
+			}
+			if(proposal.clubId.equals(req.session.userId)) {
+				return res.json({ success: true, message: "Proposal Deleted" });
+			}
+			else {
+				return res.status(403).send({ success: false, message: "Forbidden" });
+			}
+		});
```

Comment:
The proposal is being deleted before its ownership is compared to the user. First find the proposal, then only upon successful comparison delete it.